CMakeLists.txt

```diff
@@ -224,6 +224,13 @@ option(CURL_DISABLE_TFTP "disables TFTP" OFF)
 mark_as_advanced(CURL_DISABLE_TFTP)
 option(CURL_DISABLE_VERBOSE_STRINGS "disables verbose strings" OFF)
 mark_as_advanced(CURL_DISABLE_VERBOSE_STRINGS)
+option(CURL_DISABLE_LIBPROXY "disables Libproxy support" OFF)
```

Comment:
This should default to `ON` since libproxy license is LGPL. It's possible users would likely not know they were compiling in incompatibly licensed code.

Reply:
> compiling in incompatibly licensed code

What are the implications of this? Distros will not be allowed to make use of this?

Why is it being linked together at all then if it's incompatibly licensed?

Reply:
It's not incompatibly-licensed. The fact that the licenses are not the same does not make them incompatible. libproxy is fine except in environments where LPGLv2+ is banned (in which case, common dependencies like GLib are also banned and most of desktop Linux is off-limits).

Reply:
libproxy being LGPL will make a problem for some, no doubt. We already support GnuTLS which also is LGPL, and for the same reason is maybe not a popular choice in numerous commercial and embedded solutions. (libcurl does not use GLib and in fact many users of libcurl don't use GLib elsewhere either)

I believe we should at least start out with *not* enabling it by default to get people to opt-in first to get usage, feedback and reports back first before we consider it enabled by default (if found).

Reply:
Optionally, it could be enabled by default when building curl as a shared library but opt in when building a static one.

Reply:
No, that would be a terrible and confusing way to do it. Besides, most builds of libcurl build both a dynamic and a static lib at once. The person who links an application with the library then decides which one to use.